VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+  significant white spaces in a string makes normalization more complicated and
+  may be a source of confusion if you remove the spaces from the string. 
+  ``vers`` avoids the ambiguity of spaces by ignoring them.
+
+- The advanced range syntax has grown to be rather complex using hyphen ranges,
+  stars ranges, carets and tilde constructs that are all tied to the JavaScript
+  and npm ways of handling versions in their ecosystem and are bound furthermore
+  to the semver semantics and its npm implementation. These are not readily
+  reusable elsewhere. The multiple comparators and modifiers make the notation
+  grammar more complex to parse and process for a machine and harder to read for
+  human.
+
+Notations that are directly derived from node-semver as used in Rust and PHP
+Composer have the same issues.
+
+
+Why not use Python PEP-0440 ranges?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+See:
+
+- https://www.python.org/dev/peps/pep-0440/#version-specifiers
+
+The Python pep-0440 "Version Identification and Dependency Specification"
+provides a comprehensive specification for Python package versioning and a
+notation for "version specifiers" to express the version constraints of
+dependencies.
+
+This specification is similar to this ``vers`` spec, with more operators and
+aspects specific to the versions used only in the Python ecosystem.
+
+- In particular pep-0440 uses tilde, triple equal and wildcard star operators
+  that are specific to how two Python versions are compared.
+
+- The comma separator between constraints is a logical "AND" rather than an
+  "OR". The "OR" does not exist in the syntax making some version ranges
+  harder to express, in particular for vulnerabilities that may affect several
+  exact versions or ranges for multiple parallel release branches. Ranges such as
+  "Django 1.2 or later, or Django 2.2 or later or Django 3.2 or later" are
+  difficult to express without an "OR" logic.
+
+
+Why not use Rubygems requirements notation?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+See:
+
+- https://guides.rubygems.org/patterns/#declaring-dependencies
+
+The Rubygems specification suggests but does not enforce using semver. It uses
+operators similar to the ``node-semver`` spec with the different of the "~>"
+aka. pessimistic operator vs. a plain "~" tilde used in node-semver.  This
+operator implies some semver-like versioning, yet gem version are not strictly
+semver. This makes the notation complex to implement and impractical to reuse
+in places that do not use the same Ruby-specific semver-like semantics.
+
+
+Why not use fewer comparators with only =, >= and <?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For instance, the OSV schema adopts a reduced set of only three comparators:
+
+- "=" is implied when used to enumerate vulnerable versions
+- ">=" (greater or equal) is for the version that introduces a vulnerability
+- "<" (lesser) is for the version that fixes a vulnerability
```

Comment:
Note: OSV now supports the equivalent of a "<=", and this brings it to parity with CVE 5.0 (which also does not support >).

We still have strong objections against >, because it can lead to very misleading ranges (see https://github.com/ossf/osv-schema/issues/31#issuecomment-1036266822). GitHub also needed this initially, but they were able to remove this completely from their DB and no longer need this: https://github.com/github/advisory-database/issues/19#issuecomment-1414558129